cdap-ui/app/hydrator/services/hydrator-service.js

```diff
@@ -434,6 +434,15 @@ class HydratorPlusPlusHydratorService {
       userRuntimeArgumentsMap
     };
   }
+
+  getMacrosWithNonEmptyValues(macrosMap) {
+    let macrosMapCopy = Object.assign({}, macrosMap);
+    let {keysWithMissingValue} = this.myHelpers.objHasMissingValues(macrosMapCopy);
+    keysWithMissingValue.forEach(key => {
```

Comment:
This would be a problem if this is `false` right? `objHasMissingValues` returns false if macrosMapCopy is empty which could be the case for realtime pipelines